allennlp/models/archival.py

```diff
@@ -64,13 +69,16 @@ def load_model(archive_file: str, cuda_device: int = -1) -> Model:
     # Load config
     config = Params.from_file(os.path.join(tempdir, _CONFIG_NAME))
 
-    # Instantiate model
-    model = Model.load(config,
+    # Load vocabulary
+    vocab = Vocabulary.from_files(os.path.join(tempdir, "vocabulary"))
```

Comment:
originally I was going to do this. here's the problem. `model._vocab` is just a convention, we have no way of enforcing it. so what I was going to do was give `Model` a constructor that takes `vocab` and make every subclass constructor call it. then `model.vocab` would be guaranteed to make sense. I'm still happy to do that, but this seemed simpler (and more inefficient). thoughts?